src/index.ts

```diff
@@ -83,72 +233,84 @@ export class PineTest extends PinejsClientCore<unknown> {
-	public patch(
-		params: Params,
-	): PromiseResult<ResolvableReturnType<Super['patch']>> {
+	public patch<T extends Resource = AnyResource>(
+		params: {
+			resource?: undefined;
+			url: NonNullable<Params<T>['url']>;
+		} & Params<T>,
+	): PromiseResult<void>;
+	public patch(params: Params<AnyResource>): PromiseResult<void> {
 		return super.patch(
-			params as Parameters<Super['patch']>[0],
-		) as PromiseResult<ResolvableReturnType<Super['patch']>>;
+			params as Parameters<PinejsClientCore<unknown, Model>['patch']>[0],
+		) as PromiseResult<
+			ResolvableReturnType<PinejsClientCore<unknown, Model>['patch']>
+		>;
 	}
 
-	public post(
-		params: { resource: NonNullable<Params['resource']> } & Params,
-	): PromiseResult<ResolvableReturnType<Super['post']>>;
+	public post<TResource extends StringKeyOf<Model>>(
+		params: { resource: TResource } & Params<Model[TResource]>,
+	): PromiseResult<PickDeferred<Model[TResource]['Read']>>;
 	/**
 	 * @deprecated POSTing via `url` is deprecated
 	 */
-	public post(
-		params: { resource?: undefined; url: NonNullable<Params['url']> } & Params,
-	): PromiseResult<ResolvableReturnType<Super['post']>>;
-	public post(
-		params: Params,
-	): PromiseResult<ResolvableReturnType<Super['post']>> {
-		return super.post(params as Parameters<Super['post']>[0]) as PromiseResult<
-			ResolvableReturnType<Super['post']>
+	public post<T extends Resource = AnyResource>(
+		params: {
+			resource?: undefined;
+			url: NonNullable<Params<T>['url']>;
+		} & Params<T>,
+	): PromiseResult<AnyObject>;
+	public post(params: Params<AnyResource>): PromiseResult<AnyObject> {
+		return super.post(
+			params as Parameters<PinejsClientCore<unknown, Model>['post']>[0],
+		) as PromiseResult<
+			ResolvableReturnType<PinejsClientCore<unknown, Model>['post']>
 		>;
 	}
 
-	public delete(
-		params: { resource: NonNullable<Params['resource']> } & Params,
-	): PromiseResult<ResolvableReturnType<Super['delete']>>;
+	public delete<TResource extends StringKeyOf<Model>>(
+		params: { resource: TResource } & Params<Model[TResource]>,
+	): PromiseResult<void>;
 	/**
 	 * @deprecated DELETEing via `url` is deprecated
 	 */
-	public delete(
-		params: { resource?: undefined; url: NonNullable<Params['url']> } & Params,
-	): PromiseResult<ResolvableReturnType<Super['delete']>>;
-	public delete(
-		params: Params,
-	): PromiseResult<ResolvableReturnType<Super['delete']>> {
+	public delete<T extends Resource = AnyResource>(
+		params: {
+			resource?: undefined;
+			url: NonNullable<Params<T>['url']>;
+		} & Params<T>,
+	): PromiseResult<void>;
+	public delete(params: Params<AnyResource>): PromiseResult<void> {
 		return super.delete(
-			params as Parameters<Super['delete']>[0],
-		) as PromiseResult<ResolvableReturnType<Super['delete']>>;
+			params as Parameters<PinejsClientCore<unknown, Model>['delete']>[0],
+		) as PromiseResult<
+			ResolvableReturnType<PinejsClientCore<unknown, Model>['delete']>
+		>;
```

Comment:
The type of the result comes from the return type of `PinejsClientCore<unknown, Model>['delete']` so the current form is accurate to what it will actually be, yes we expect that to resolve to `PromiseResult<void>` but if it does not for some reason (eg pinejs-client-core changes the return type) then typescript would flag it up as an error so we'd get a compile time warning with the current version but not with the `as PromiseResult<void>` version